test/api.test.ts

```diff
@@ -0,0 +1,352 @@
+import "jest";
+import { aws_apigateway, aws_dynamodb, IResolvable, Stack } from "aws-cdk-lib";
+import {
+  AwsApiIntegration,
+  MockApiIntegration,
+  Function,
+  BaseApiIntegration,
+  ExpressStepFunction,
+  Table,
+  $AWS,
+  ApiGatewayInput,
+} from "../src";
+
+let stack: Stack;
+let func: Function<any, any>;
+beforeEach(() => {
+  stack = new Stack();
+  func = new Function(stack, "F", (p) => {
+    return p;
+  });
+});
+
+test("mock integration with object literal", () => {
+  const api = new aws_apigateway.RestApi(stack, "API");
+
+  const method = getCfnMethod(
+    new MockApiIntegration(
+      {
+        httpMethod: "GET",
+        resource: api.root,
+      },
+      ($input) => ({
+        statusCode: $input.params("code") as number,
+      }),
+      {
+        200: () => ({
+          response: "OK",
+        }),
+        500: () => ({
+          response: "BAD",
+        }),
+      }
+    )
+  );
+
+  expect(method.httpMethod).toEqual("GET");
+  expect(method.integration.requestTemplates).toEqual({
```

Comment:
use snapshots instead of hand written? these are a pain to keep up to date as we refactor. (ex: async)